stackslib/src/chainstate/stacks/db/mod.rs

```diff
@@ -292,45 +292,20 @@ pub struct DBConfig {
 
 impl DBConfig {
     pub fn supports_epoch(&self, epoch_id: StacksEpochId) -> bool {
+        let version_u32: u32 = self.version.parse().unwrap_or_else(|e| {
+            error!("Failed to parse Stacks chainstate version as u32: {e}");
+            0
+        });
         match epoch_id {
             StacksEpochId::Epoch10 => true,
-            StacksEpochId::Epoch20 => {
-                self.version == "1"
-                    || self.version == "2"
-                    || self.version == "3"
-                    || self.version == "4"
-                    || self.version == "5"
-            }
-            StacksEpochId::Epoch2_05 => {
-                self.version == "2"
-                    || self.version == "3"
-                    || self.version == "4"
-                    || self.version == "5"
-            }
-            StacksEpochId::Epoch21 => {
-                self.version == "3" || self.version == "4" || self.version == "5"
-            }
-            StacksEpochId::Epoch22 => {
-                self.version == "3" || self.version == "4" || self.version == "5"
-            }
-            StacksEpochId::Epoch23 => {
-                self.version == "3" || self.version == "4" || self.version == "5"
-            }
-            StacksEpochId::Epoch24 => {
-                self.version == "3" || self.version == "4" || self.version == "5"
-            }
-            StacksEpochId::Epoch25 => {
-                self.version == "3"
-                    || self.version == "4"
-                    || self.version == "5"
-                    || self.version == "6"
-            }
-            StacksEpochId::Epoch30 => {
-                self.version == "3"
-                    || self.version == "4"
-                    || self.version == "5"
-                    || self.version == "6"
-            }
+            StacksEpochId::Epoch20 => version_u32 >= 1 && version_u32 <= 5,
+            StacksEpochId::Epoch2_05 => version_u32 >= 2 && version_u32 <= 5,
+            StacksEpochId::Epoch21 => version_u32 >= 3 && version_u32 <= 5,
+            StacksEpochId::Epoch22 => version_u32 >= 3 && version_u32 <= 5,
+            StacksEpochId::Epoch23 => version_u32 >= 3 && version_u32 <= 5,
+            StacksEpochId::Epoch24 => version_u32 >= 3 && version_u32 <= 5,
```

Comment:
Don't these upper bounds mean that the node will panic if it syncs from genesis when the version is `6`?

Reply:
Hmm, it would crash if you stopped and resumed genesis sync.